src/structural_properties.c

```diff
@@ -5885,53 +5908,52 @@ int igraph_avg_nearest_neighbor_degree(const igraph_t *graph,
   igraph_vit_t vit;
   igraph_vector_t my_knn_v, *my_knn=knn;
   igraph_vector_t deg;
-  long int maxdeg;
+  igraph_integer_t maxdeg;
   igraph_vector_t deghist;
   igraph_real_t mynan=IGRAPH_NAN;
   igraph_bool_t simple;
 
   IGRAPH_CHECK(igraph_is_simple(graph, &simple));
   if (!simple) {
-    IGRAPH_ERROR("Average nearest neighbor degree Works only with "
+    IGRAPH_ERROR("Average nearest neighbor degree works only with "
 		 "simple graphs", IGRAPH_EINVAL);
   }
 
-  if (weights) { 
-    return igraph_i_avg_nearest_neighbor_degree_weighted(graph, vids, knn, knnk,
-							 weights);
+  if (weights) {
+    return igraph_i_avg_nearest_neighbor_degree_weighted(graph, vids,
+        mode, neighbor_degree_mode, knn, knnk, weights);
   }
-  
+
   IGRAPH_CHECK(igraph_vit_create(graph, vids, &vit));
   IGRAPH_FINALLY(igraph_vit_destroy, &vit);
   no_vids=IGRAPH_VIT_SIZE(vit);
-  
+
   if (!knn) {
     IGRAPH_VECTOR_INIT_FINALLY(&my_knn_v, no_vids);
     my_knn=&my_knn_v;
   } else {
     IGRAPH_CHECK(igraph_vector_resize(knn, no_vids));
   }
-  
+
   IGRAPH_VECTOR_INIT_FINALLY(&deg, no_of_nodes);
   IGRAPH_CHECK(igraph_degree(graph, &deg, igraph_vss_all(),
-			     /*mode=*/ IGRAPH_ALL, /*loops*/ 1));
-  maxdeg=(long int) igraph_vector_max(&deg);
+			     neighbor_degree_mode, IGRAPH_LOOPS));
+  igraph_maxdegree(graph, &maxdeg, igraph_vss_all(), mode, IGRAPH_LOOPS);
   IGRAPH_VECTOR_INIT_FINALLY(&neis, maxdeg);
   igraph_vector_resize(&neis, 0);
 
   if (knnk) {
-    IGRAPH_CHECK(igraph_vector_resize(knnk, maxdeg));
+    IGRAPH_CHECK(igraph_vector_resize(knnk, (long int)maxdeg));
     igraph_vector_null(knnk);
-    IGRAPH_VECTOR_INIT_FINALLY(&deghist, maxdeg);
+    IGRAPH_VECTOR_INIT_FINALLY(&deghist, (long int)maxdeg);
   }
 
   for (i=0; !IGRAPH_VIT_END(vit); IGRAPH_VIT_NEXT(vit), i++) {
     igraph_real_t sum=0.0;
     long int v=IGRAPH_VIT_GET(vit);
-    long int nv=(long int) VECTOR(deg)[v];
-    IGRAPH_CHECK(igraph_neighbors(graph, &neis, (igraph_integer_t) v,
-				  IGRAPH_ALL));
-    for (j=0; j<nv; j++) { 
+    IGRAPH_CHECK(igraph_neighbors(graph, &neis, (igraph_integer_t) v, mode));
+    long int nv=igraph_vector_size(&neis);
```

Comment:
The declaration of `nv` should be moved to the top of the block; otherwise the CI build will fail on Windows with Python 2.7.

Reply:
A recurrent mistake of mine :)

Is there any way to force `gcc` to conform to this declaration standard, without forcing also additional `C89` features that are compatible with `VC9++` to compile for Python 2.7 (e.g. `//` comment style)?

Reply:
I don't think so; let me know if you find a way ;)